### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by arnaudforaison**
Update all dependencies, so :

- remove storybook & design system dev dependencies into root package.json
- move dependencies specific for package into it
- update versions of dev dependencies
- fix script with update dependencies breaking changes

In this PR, storybook is broken because some dependencies removed from root package.json are not move into storybook.
#700 migrate storybook to new version I doesn't want to interfer with this.

### CODE REVIEW

**Comment by xballoy** [comment]
The build is KO: 'Cannot find module 'babel-runtime/helpers/*'.
Otherwise great job! 👍

**Comment by arnaudforaison** [comment]
Yes I know. I'm investigate because it's always on test in CI, not in local. 
Pipeline seems to run extract tests

**Comment by xballoy** [comment]
I don't think its related but have also have this issue in my IDE: 
```
Error: Cannot find module 'eslint-config-react-app'
Referenced from: /Users/A181NI/Projects/react-toolkit/.eslintrc
```

File: package.json [hunk 1]
```diff
@@ -81,84 +62,30 @@
...
-    "simple-git": "1.70.0",
-    "style-loader": "0.21.0",
-    "stylelint": "9.2.0",
-    "stylelint-config-standard": "18.2.0",
-    "svg-inline-loader": "^0.8.0",
-    "svg-sprite": "1.5.0",
-    "ts-jest": "^23.10.5",
-    "ts-loader": "^4.4.2",
+    "ts-jest": "^26.4.1",
     "tslint": "~5.20.0",
```
Comment (xballoy) [inline package.json:79]: Can you cant it to the "^6.1.3" (currently installed)

**Review by xballoy — APPROVED** [review]
Great job

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -73,12 +73,11 @@ npm test
...
 
 By default, `npm test` also runs the linter.
-You can skip this by calling `jest` directly:
+You can add args to `npm test` command with `--`:
 
 ```sh
-$ npm jest
-$ npm jest --watch
-$ npm jest --config jest.config.json
+$ npm test -- --watch
```
Comment (samuel-gomez-axa) [inline CONTRIBUTING.md:79]: we can also use "npm t" :)

File: azure-pipelines.yml [hunk 3]
```diff
@@ -42,15 +42,15 @@ steps:
...
 
-  - script: npm i
-    displayName: 'npm install'
-
-  - script: npm ci --prefix storybook/storybook
+  - script: cd storybook/storybook && npm ci
     displayName: 'npm install storybook'
 
-  - script: npm ci --prefix storybook/design-system
+  - script: cd storybook/design-system && npm ci
```
Comment (samuel-gomez-axa) [inline azure-pipelines.yml:48]: why do you change this ? this did'nt work ?

File: packages/Layout/header/src/NavBar/NavBar.js [hunk 4]
```diff
@@ -39,10 +39,6 @@ class NavBar extends Component {
     };
   }
 
-  componentWillReceiveProps(nextProps) {
```
Comment (samuel-gomez-axa) [inline packages/Layout/header/src/NavBar/NavBar.js:42]: is there any impact to delete this lifecycle method ?

Comment (samuel-gomez-axa) [reply packages/Layout/header/src/NavBar/NavBar.js:42]: should be replace by getDerivedStateFromProps, what you think ?

File: azure-pipelines.yml (hunk 3 above)
Comment (arnaudforaison) [reply azure-pipelines.yml:48]: I think it's was the cause of bad dependencies installation. But not.
But it's more simple command and this use case of prefix seems to not do what we want

File: packages/Layout/header/src/NavBar/NavBar.js (hunk 4 above)
Comment (arnaudforaison) [reply packages/Layout/header/src/NavBar/NavBar.js:42]: I use getDerivedStateFromProps but doesn't work (tests fail). But are you agree that if props change, component will be rerender & state with init prop will be updated ?

File: azure-pipelines.yml (hunk 3 above)
Comment (samuel-gomez-axa) [reply azure-pipelines.yml:48]: ok

File: packages/Layout/header/src/NavBar/NavBar.js (hunk 4 above)
Comment (samuel-gomez-axa) [reply packages/Layout/header/src/NavBar/NavBar.js:42]: you're right, i think is useless here : https://fr.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

**Review by samuel-gomez-axa — APPROVED** [review]

**Pull request merged**